lib/src/Base/Stat/LatentVariableModel.cxx

```diff
@@ -0,0 +1,441 @@
+  for (UnsignedInteger i = 0; i < inputDimension_; ++ i)
+  {
+    if (!(parameter[index] > 0.0))
+      throw InvalidArgumentException(HERE) << "In LatentVariableModel::setFullParameter, the component " << index << " of scale is non positive";
+    scale_[i] = parameter[index];
+    ++ index;
+  }
+  // Second the amplitude parameter
+  for (UnsignedInteger i = 0; i < outputDimension_; ++ i)
+  {
+    if (!(parameter[index] > 0.0))
+      throw InvalidArgumentException(HERE) << "In LatentVariableModel::setFullParameter, the component " << index << " of amplitude is non positive";
+    amplitude_[i] = parameter[index];
+    ++ index;
+  }
+
+  // Update the latent covariance model
+  latCovMod_.setAmplitude(amplitude_);
+  latCovMod_.setScale(Point(latent_dim_,scale_[0]));
+
+  // Third the latent variable coordinates
+  Point activeLatentVariables = Point(activeLatentCoordinateDim_,1.0);
+  for (UnsignedInteger i = 0; i < activeLatentCoordinateDim_; ++ i)
+  {
+    activeLatentVariables[i] = parameter[index];
+    ++ index;
+  }
+  setLatentVariables(activeLatentVariables);
+
+
+}
+
+Point LatentVariableModel::getFullParameter() const
+{
+  // Get the generic parameter
+  Point parameter(CovarianceModelImplementation::getFullParameter());
+  // Add the specific one
+  parameter.add(activeLatentVariables_);
+  return parameter;
+}
+
+Description LatentVariableModel::getFullParameterDescription() const
+{
+  // Description of the generic parameter
+  Description description(CovarianceModelImplementation::getFullParameterDescription());
+  // Description of the specific parameter
+  for (UnsignedInteger i = 0; i < activeLatentCoordinateDim_; ++i)
+    description.add(OSS() << "activeLatentVariable_" << i);
+  return description;
+}
+
+/* String converter */
+String LatentVariableModel::__repr__() const
+{
+  OSS oss;
+  oss << "class=" << LatentVariableModel::GetClassName()
+      << " scale=" << scale_
+      << " amplitude=" << amplitude_
+      << " activeLatentVariables=" << activeLatentVariables_;
+  return oss;
+}
+
+/* String converter */
+String LatentVariableModel::__str__(const String & ) const
+{
+  OSS oss;
+  oss << LatentVariableModel::GetClassName()
+      << "(scale=" << scale_.__str__()
+      << ", amplitude=" << amplitude_.__str__()
+      << ", activeLatentVariables=" << activeLatentVariables_.__str__()
+      << ")";
+  return oss;
+}
+
+
+/* latentVariables accessor */
+void LatentVariableModel::setLatentVariables(const Point & latentVariablesCoordinates)
+{
+  if (!(latentVariablesCoordinates.getSize() == activeLatentCoordinateDim_))
+    throw InvalidArgumentException(HERE) << "In LatentVariableModel::setLatentVariables, samples have incompatible size. Sample size = " << latentVariablesCoordinates.getSize()
+                                         << " whereas expected size = " << activeLatentCoordinateDim_;
+  // Set the active variables
+  activeLatentVariables_ = latentVariablesCoordinates;
```

Comment:
the name `activeLatentVariables_` is confusing

Reply:
@sofianehaddad What name would you suggest?